### PR TITLE
Made marketplace compatible with Firefox browser

### PR DESCRIPTION
**Pull request by AshanKooragoda**
bug is not added yet. since that no jira ticket is attached.

### CODE REVIEW

File: symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/templates/ohrmAddonsSuccess.php [hunk 1]
```diff
@@ -78,7 +78,7 @@
                                            echo 'background-color: #808080;';
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/templates/ohrmAddonsSuccess.php]: Seems hardcoded style here

Comment (AshanKooragoda) [reply symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/templates/ohrmAddonsSuccess.php]: It is depending on the addonl list pass to success file. I can create class and add PHP to make it available when if condition is there. Is it ok?

File: symfony/plugins/orangehrmMarketPlacePlugin/web/js/ohrmAddonSuccess.js [hunk 2]
```diff
@@ -40,7 +44,7 @@ $(document).ready(function () {
             data: {installAddonID: installId},
             url: installUrl, success: function (result) {
                 if (result === '"Success"') {
-                    $('#loading').attr('class', '')
+                    $('#loading').attr('class', '');
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmMarketPlacePlugin/web/js/ohrmAddonSuccess.js]: $('#loading').removeClass()

File: symfony/plugins/orangehrmMarketPlacePlugin/web/js/ohrmAddonSuccess.js [hunk 3]
```diff
@@ -111,7 +115,10 @@ $(document).ready(function () {
                         $('#buyBtn' + buyNowId).attr('value', 'Requested').prop("disabled", true).css('background-color', '#808080');
                         $('#disable-screen').attr('class', '');
                         $('#messege_div').show(0).delay(2000).fadeOut(1000);
-                    } else {
+                    } else if (result === '"Network Error"') {
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmMarketPlacePlugin/web/js/ohrmAddonSuccess.js]: OOPS  .. this is wrong

File: symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/templates/ohrmAddonsSuccess.php (hunk 1 above)
Comment (AshanKooragoda) [reply symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/templates/ohrmAddonsSuccess.php]: corrected

File: symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/getAddonDescriptionAPIAction.class.php [hunk 4]
```diff
@@ -24,37 +24,43 @@ class getAddonDescriptionAPIAction extends baseAddonAction
...
+            $addonList = $this->getAddons();
+            foreach ($addonList as $addon) {
+                if ($addon['id'] == $addonId) {
+                    $addonDesCriptionURL = $addon['links']['desc'];
+                }
             }
+            $addonDescription = $this->getDescription($addonDesCriptionURL);
+            $this->addonDescription = base64_decode($addonDescription);
+        } catch (GuzzleHttp\Exception\ConnectException $e) {
+            echo json_encode('0');
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/getAddonDescriptionAPIAction.class.php]: Are we showing "No internet connection" msg based on this error?

File: symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/installAddonAPIAction.class.php [hunk 5]
```diff
@@ -30,60 +30,55 @@ class installAddonAPIAction extends baseAddonAction
...
-            if ($addon['id'] == $addonId) {
-                $addonDetail = $addon;
-                $addonURL = $addon['links']['file'];
+        try {
+            $addonList = $this->getAddons();
+            $data = $request->getParameterHolder()->getAll();
+            $addonId = $data['installAddonID'];
+            $addonURL = null;
+            $addonDetail = null;
+            foreach ($addonList as $addon) {
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/installAddonAPIAction.class.php]: Why looping through all add-ons to get one file?

File: symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/ohrmBuyNowAPIAction.class.php [hunk 6]
```diff
@@ -26,44 +26,46 @@ class ohrmBuyNowAPIAction extends baseAddonAction
...
+            foreach ($addonList as $addon) {
+                if ($addon['id'] == $addonId) {
+                    $addonDetail = $addon;
+                }
             }
+            $result = $this->buyNow($data, $addonDetail);
+            echo json_encode($result);
+            return sfView::NONE;
+        } catch (GuzzleHttp\Exception\ConnectException $e) {
+            echo json_encode('0');
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/ohrmBuyNowAPIAction.class.php]: Define ohrm error codes as constant 
0 => 'ERROR_CODE_NO_CONNECTION'
1 => 'ERROR_CODE_EXCEPTION'

File: symfony/web/themes/default/css/main.css [hunk 7]
```diff
@@ -1708,10 +1708,12 @@ div.personalDetails {
 
 #MP_link {
 	font: inherit;
-	font-size: 100%;
+	font-size: 105%;
```
Comment (samanthajayasinghe) [inline symfony/web/themes/default/css/main.css]: are you sure about 105% ?

File: symfony/plugins/orangehrmMarketPlacePlugin/lib/dao/MarketplaceDao.php [hunk 8]
```diff
@@ -91,7 +91,7 @@ public function uninstallAddon($addonId)
             return $q->execute();
             // @codeCoverageIgnoreStart
         } catch (Exception $e) {
-            throw new DaoException($e->getMessage(), $e->getCode(), $e);
+            throw $e;
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmMarketPlacePlugin/lib/dao/MarketplaceDao.php]: why catch the exception if you throw same ?

File: symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/baseAddonAction.class.php [hunk 9]
```diff
@@ -22,6 +22,15 @@
  */
 abstract class baseAddonAction extends sfAction
 {
+    /**
+     * No Network Error Message
+     */
+    const NO_NETWORK_ERR_MESSAGE = "please connect to view ALL AVAILABLE ADDONS";
```
[the remark this answers is not in the document]
Comment (AshanKooragoda) [reply symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/baseAddonAction.class.php]: yes . this was added as a Jira ticket from QA asking to change Message as above.  I informed about this to kavitha. 
https://jira.orangehrm.com/browse/OHRM-190

File: symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/getAddonDescriptionAPIAction.class.php (hunk 4 above)
Comment (AshanKooragoda) [reply symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/getAddonDescriptionAPIAction.class.php]: yes

Comment (AshanKooragoda) [reply symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/getAddonDescriptionAPIAction.class.php]: I add it in doc comment

File: symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/installAddonAPIAction.class.php (hunk 5 above)
Comment (AshanKooragoda) [reply symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/installAddonAPIAction.class.php]: can you suggest a better way?

File: symfony/plugins/orangehrmMarketPlacePlugin/lib/dao/MarketplaceDao.php (hunk 8 above)
Comment (AshanKooragoda) [reply symfony/plugins/orangehrmMarketPlacePlugin/lib/dao/MarketplaceDao.php]: corrected.

File: symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/baseAddonAction.class.php (hunk 9 above)
[the remark this answers is not in the document]
Comment (AshanKooragoda) [reply symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/baseAddonAction.class.php]: Sry I misunderstood the jira reported by QA. Corrected it.

File: symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/baseAddonAction.class.php [hunk 10]
```diff
@@ -22,6 +22,23 @@
...
+     */
+    const NO_NETWORK_ERR_MESSAGE = "Please connect to the internet to view the available add-ons.";
+    /**
+     * Marketplace middlewere Error Message
+     */
+    const MP_MIDDLEWERE_ERR_MESSAGE = "Error Occur Please try again later";
+    /**
+     * network error code
+     */
+    const ERROR_CODE_NO_CONNECTION = '0';
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/baseAddonAction.class.php]: should be int

File: symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/baseAddonAction.class.php [hunk 11]
```diff
@@ -22,6 +22,23 @@
...
+    /**
+     * Marketplace error message code
+     */
+    const ERROR_CODE_EXCEPTION = '1';
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/baseAddonAction.class.php]: should be int

File: symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/installAddonAPIAction.class.php (hunk 5 above)
Comment (samanthajayasinghe) [reply symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/installAddonAPIAction.class.php]: can't you call the api with add-on id ? this leads to performance issue if there are big number of add-on list

File: symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/baseAddonAction.class.php (hunk 10 above)
Comment (AshanKooragoda) [reply symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/baseAddonAction.class.php]: done

File: symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/baseAddonAction.class.php (hunk 11 above)
Comment (AshanKooragoda) [reply symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/baseAddonAction.class.php]: done

File: symfony/plugins/orangehrmMarketPlacePlugin/lib/manager/APIManagerService.php [hunk 12]
```diff
@@ -276,4 +277,16 @@ public function getConfigService()
...
+     * @return string
+     */
+    public function getVersion()
+    {
+        if (@include_once sfConfig::get('sf_root_dir') . "/../lib/confs/sysConf.php") {
+            $version = new sysConf();
+            $version = $version->getVersion();
+        }
+        return $version;
+    }
```
Comment (AshanKooragoda) [inline symfony/plugins/orangehrmMarketPlacePlugin/lib/manager/APIManagerService.php]: I get this from OS existing code. needs to know if it is fine. This is regarding getting OS version

**Comment by samanthajayasinghe** [comment]
@AshanKooragoda  you need to rebase this PR before merging ..

**Pull request merged**